app/flux/components/Checkout.react.js

```diff
@@ -0,0 +1,96 @@
+var React = require('react');
+
+var { State } = require('react-router');
+var StoreAPI = require('../API/StoreAPI');
+
+var UserStore = require('../stores/UserStore');
+
+class Checkout extends React.Component{
+    constructor (props){
+        super(props);
+
+        // Only lifecycle methods receive the inheritance from React.Component.
+        this._onChange = this._onChange.bind(this);
+        
+        this.state = {
+            stripe: null,
+            training: null,
+            user: UserStore.user        
+        };
+    }
+
+    componentDidMount (){
+        UserStore.addChangeListener(this._onChange);
+
+        if (!this.state.user && UserStore.isPending){
+            setTimeout(() => {
+                if (!this.state.user){
+                    this.context.router.transitionTo('signup');    
+                }
+            }, 800).bind(this);
+        } else if (!this.state.user){
+            this.context.router.transitionTo('signup');
+        }
+
+        var trainingId = this.context.router.getCurrentParams().id;
+        StoreAPI.get.checkout(trainingId).then(response => {
+            /**
+             * stripe: string,
+             * training: object
+             */
+            var { stripe, training } = response;
+            this.setState({
+                stripe: stripe,
+                training: training
+            });
+        });
+    }
+
+    componentWillUnmount (){
+        UserStore.removeChangeListener(this._onChange);
+    }
+
+    render (){
+        var { stripe, training } = this.state;
+
+        if(!training) {
+            return <div></div>;
+        }
+
+        return (
+            <main className="checkout">
+                <form action={'/api/buy/' + training.id} method="post">
+                    { training.cost ? 
```

Comment:
This should be changed and is highly confusing, just create a function that handles each case to make this more clear.